src/slider/slider.js

```diff
@@ -5,200 +5,141 @@ This source code is licensed under the MIT license found in the
-      });
-    }
-  }
-
   getSharedProps() {
-    const {range, value} = this.props;
-    const {currentThumb} = this.state;
+    const {disabled, step, min, max, values}: PropsT = this.props;
     return {
-      $value: value,
-      $range: range,
-      $max: getTickValue(range[range.length - 1]),
-      $min: getTickValue(range[0]),
-      $currentThumb: currentThumb,
-      $isRange: value.length % 2 === 0,
+      $disabled: disabled,
+      $step: step,
+      $min: min,
+      $max: max,
+      $values: limitValues(values),
     };
   }
 
-  static clampIt(value: number, max: number, min: number) {
-    return Math.max(min, Math.min(value, max));
-  }
-  static scale(x: number, distanceX: number, distanceY: number) {
-    return (x * distanceY) / distanceX;
-  }
-
   render() {
-    const {overrides = {}, range, value} = this.props;
-    const {thumbRefs} = this.state;
+    const {overrides = {}, min, max, step, onChange, disabled} = this.props;
+    const values = limitValues(this.props.values);
     const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
-    const [Axis, axisProps] = getOverrides(overrides.Axis, StyledAxis);
-    const [AxisRange, axisRangeProps] = getOverrides(
-      overrides.AxisRange,
-      StyledAxisRange,
+    const [Track, trackProps] = getOverrides(overrides.Track, StyledTrack);
+    const [InnerTrack, innerTrackProps] = getOverrides(
+      overrides.InnerTrack,
+      StyledInnerTrack,
     );
     const [Thumb, thumbProps] = getOverrides(overrides.Thumb, StyledThumb);
+    const [InnerThumb, innerThumbProps] = getOverrides(
+      overrides.InnerThumb,
+      StyledInnerThumb,
+    );
+    const [ThumbValue, thumbValueProps] = getOverrides(
+      overrides.ThumbValue,
+      StyledThumbValue,
+    );
     const [Tick, tickProps] = getOverrides(overrides.Tick, StyledTick);
     const [TickBar, tickBarProps] = getOverrides(
       overrides.TickBar,
       StyledTickBar,
     );
     const sharedProps = this.getSharedProps();
     return (
-      <Root {...sharedProps} onClick={e => this.onAxisClick(e)} {...rootProps}>
-        <Axis {...sharedProps} {...axisProps}>
-          {value.map((thumb, $index) => (
-            <React.Fragment key={$index}>
-              <AxisRange $index={$index} {...sharedProps} {...axisRangeProps} />
-              <Thumb
-                $ref={thumbRefs[$index]}
-                key={$index}
-                $index={$index}
-                onMouseDown={e => this.onThumbDown(e, $index)}
+      <Root {...sharedProps} {...rootProps}>
+        <Range
+          step={step}
+          min={min}
+          max={max}
+          values={values}
+          disabled={disabled}
+          onChange={values => onChange({values})}
+          renderTrack={({props, children, isDragged}) => (
+            <Track
+              onMouseDown={props.onMouseDown}
+              onTouchStart={props.onTouchStart}
+              $isDragged={isDragged}
+              style={props.style}
```

Comment:
What is this style prop? Should allow style overrides through regular overrides API